src/components/NcFormBoxSwitch/NcFormBoxSwitch.vue

```diff
@@ -0,0 +1,155 @@
+<!--
+  - SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<script setup lang="ts">
+import type { Slot } from 'vue'
+
+import { mdiToggleSwitch, mdiToggleSwitchOff } from '@mdi/js'
+import { watch } from 'vue'
+import NcFormBoxItem from '../NcFormBox/NcFormBoxItem.vue'
+import NcIconSvgWrapper from '../NcIconSvgWrapper/NcIconSvgWrapper.vue'
+import { createElementId } from '../../utils/createElementId.ts'
+
+/** Switch toggle model value */
+const modelValue = defineModel<boolean>({ required: true })
+
+const {
+	label = undefined,
+	description = undefined,
+	disabled = false,
+} = defineProps<{
+	/** Main label */
+	label?: string
+	/** Optional description below the label, also used for the aria-describedby */
+	description?: string
+	/** Disabled state */
+	disabled?: boolean
+}>()
+
+const emit = defineEmits<{
+	/** Switch is toggled ON */
+	enable: []
+	/** Switch is toggled OFF */
+	disable: []
+}>()
```

Comment:
I do not really get the use case, its basically duplicates the model value.
All you can do with the events you can do with `update:modelValue` as well?

Reply:
> All you can do with the events you can do with update:modelValue as well?

You can, with additional actions and less explicit.

With events:
```html
<template>
  <NcFormBoxCopyButton @enable="handleEnabling">
</template>
```

Without:
```html
<script setup>
/**
 * Handle mopdel value change
 *
 * @param newValue - new modelValue
 */
function handleChange(newValue: boolean) {
  if (newValue) {
    handleEnabling()
  }
}
</script>

<template>
  <NcFormBoxSwitch @update:modelValue="handleEnabling">
</template>
```

or

```html
<template>
  <NcFormBoxSwitch @update:modelValue="$event && handleEnabling()">
</template>
```